matron/src/oracle.c

```diff
@@ -484,136 +496,141 @@ void o_restart_audio() {
 }
 
-void o_tape_play() {
-    lo_send(ext_addr, "/tape/play", NULL);
+void o_tape_play_stop() {
+    lo_send(crone_addr, "/tape/play/stop", NULL);
 }
 
-void o_tape_pause() {
-    lo_send(ext_addr, "/tape/pause", NULL);
+
+//--- cut
+void o_cut_enable(int i, float value) {
+    lo_send(crone_addr, "/set/enabled/cut", "if", i, value);
 }
 
-void o_tape_stop() {
-    lo_send(ext_addr, "/tape/stop", NULL);
+void o_set_level_adc_cut(float value) {
+    lo_send(crone_addr, "/set/level/adc_cut", "f", value);
 }
 
-//--- aux effects controls
-// enable / disable aux fx processing
-void o_set_aux_fx_on() {
-    lo_send(crone_addr, "/set/enabled/reverb", "f", 1);
+void o_set_level_ext_cut(float value) {
+    lo_send(crone_addr, "/set/level/ext_cut", "f", value);
 }
 
-void o_set_aux_fx_off() {
-    lo_send(crone_addr, "/set/enabled/reverb", "f", 0);
+void o_set_level_cut_aux(float value) {
+    lo_send(crone_addr, "/set/level/cut_aux", "f", value);
 }
 
+void o_set_level_cut(int index, float value) {
+	lo_send(crone_addr, "/set/level/cut", "if", index, value);
+}
 
-//--- insert effects controls
-void o_set_insert_fx_on() {
-    lo_send(crone_addr, "/set/enabled/compressor", "f", 1);
+void o_set_level_cut_cut(int src, int dest, float value) {
+	lo_send(crone_addr, "/set/level/cut_cut", "iif", src, dest, value);
 }
 
-void o_set_insert_fx_off() {
-    lo_send(crone_addr, "/set/enabled/compressor", "f", 0);
+void o_set_pan_cut(int index, float value) {
+	lo_send(crone_addr, "/set/pan/cut", "if", index, value);
 }
 
-void o_set_insert_fx_mix(float value) {
-    lo_send(crone_addr, "/set/level/ins_mix", "f", value);
+void o_set_cut_param(const char* name, int voice, float value) {
+    static char buf[128];
+    sprintf(buf, "/set/param/cut/%s", name);    
+    lo_send(crone_addr, buf, "if", voice, value);
 }
 
+void o_set_level_input_cut(int src, int dst, float level) {
+    lo_send(crone_addr, "/set/level/in_cut", "iif", src, dst, level);
+}
 
-// stereo output -> aux
-void o_set_aux_fx_output_level(float value) {
-    lo_send(crone_addr, "/set/level/ext/aux", "f", value);
+void o_cut_buffer_clear_region(float start, float end) {
+    lo_send(crone_addr, "/softcut/buffer/clear", "ff", start, end);
 }
 
-// aux return -> dac
-void o_set_aux_fx_return_level(float value) {
-    lo_send(crone_addr, "/set/level/aux/dac", "f", value);
+void o_cut_buffer_clear() {
+    lo_send(crone_addr, "/softcut/buffer/clear", "");
 }
 
-///////////////////////////////
-///////////////////////////
-///// FIXME EEEEEEEE ???
+void o_cut_buffer_read(char *file, float start_src, float start_dst, float dur, int ch) {
+	lo_send(crone_addr, "/softcut/buffer/read", "sfffi", file, start_src, start_dst, dur, ch);
+} 
+
+
 
-/// FIXME: doesn't need channel count?
-// monitor mix -> aux level (stereo!)
-void o_set_aux_fx_input_level(int channel, float value) {
-    (void) channel;
-    lo_send(crone_addr, "/set/level/monitor/aux", "f", value);
+
```

Comment:
NB: i made several arbitrary decisions about OSC paths, which you should feel free to change. (e.g. if it seems clearest if paths match lua / oracle method names.)

one decision was to refer to effects by function and not by their place in the chain. this is because the parameter names are baked into the paths, and particular param names only make sense for particular effect types.

Reply:
renamed `aux` to `rev` and `ins` to `comp`

feels good to me?

https://github.com/monome/norns/pull/671/commits/4c2eab67025c7b04f00061db2b83d57dc7faae28